src/openforms/prefill/sources.py

```diff
@@ -70,17 +70,22 @@ def invoke_plugin(
             if values:
                 logevent.prefill_retrieve_success(submission, plugin, fields)
             else:
-                if (required_attribute := plugin.requires_auth) is None or (
-                    (auth_info := getattr(submission, "auth_info", None))
-                    and auth_info.attribute == required_attribute
-                ):
-                    logevent.prefill_retrieve_empty(submission, plugin, fields)
+                logevent.prefill_retrieve_empty(submission, plugin, fields)
         return fields, values
 
     invoke_plugin_args = []
     for plugin_id, field_groups in grouped_fields.items():
+        plugin = register[plugin_id]
+
+        # check if we need to run the plugin when the user is authenticated
+        if (required_attribute := plugin.requires_auth) is not None and (
+            (auth_info := getattr(submission, "auth_info", None))
+            and auth_info.attribute not in required_attribute
+        ):
+            continue
```

Comment:
I think it would be possible to simplify this because `BasePlugin` must define the `requires_auth` attribute and it's an empty collection by default.

```suggestion
        auth_info = getattr(submission, "auth_info", None)
        if auth_info and auth_info.attribute not in plugin.requires_auth:
            continue
```